File: AlyShmahell-BCompSci-UnivAQ/NavibusMercatoriis/WebUI/Regina.py
# -*- coding: utf-8 -*-
import os
import sys
from bottle import route,run,template


# table start
stable = '<table>'
# table end
etable = '</table>'
# row start
srow = '<tr>'
# row end
erow = '</tr>'
# column title
colt = '<th>{{value}}</th>'
# cell data
celld = '<td>{{value}}</td>'

# start select
sselect = '<select name="{{selectName}}">'
# end select
eselect = '</select>'
# select body
selectb = '<option value="{{optionValue}}">{{optionValue}}</option>'

class Regina(object):
	def createTable(self,headArray, bodyArray):
	
		table = stable + srow
		for i in range(len(headArray)):
			table += template(colt, value = headArray[i])
		table += erow
		
		i=0
		while i < len(bodyArray):
			table+=srow
			for j in range(i,i+len(headArray)):
				table += template(celld, value = bodyArray[j])
			i += len(headArray)
			table += erow
		table += etable
		
		return table

	def createSelect(self, name, valuesArray):
		select = template(sselect, selectName = name)
		for i in range(len(valuesArray)):
			select += template(selectb, optionValue = valuesArray[i])
		select += eselect
		return select
		
		
@route('/')
def hello():
	rg = Regina()
	return rg.createTable(['title1','title2'],['data1','data2']) + rg.createSelect(4,['Hala','cow','Aly','Donkey'])
    
if __name__ == '__main__':
	run(host='localhost', port=8080, debug=True)

File: AlyShmahell-BCompSci-UnivAQ/FCCR/src/site/includes/Regina/Regina.php
<?php 
namespace Regina;
class Regina
{
    public static function createTable($headArray,$bodyArray) {
        $tableStart = file_get_contents(__DIR__ . "/templates/table/tableStart.html");
        $tableEnd = file_get_contents(__DIR__ . "/templates/table/tableEnd.html");
        $rowStart = file_get_contents(__DIR__ . "/templates/table/rowStart.html");
        $rowEnd = file_get_contents(__DIR__ . "/templates/table/rowEnd.html");
        $columnTitle = file_get_contents(__DIR__ . "/templates/table/columnTitle.html");
        $cellData = file_get_contents(__DIR__ . "/templates/table/cellData.html");
    
        $table = $tableStart;
        $table = $table. $rowStart;
        for($i = 0; $i < count($headArray); $i++)
        {
            $table = $table. $columnTitle;
            $table = str_replace("{{value}}", $headArray[$i], $table);
        }
        $table = $table. $rowEnd;
    
    
        for($i = 0; $i < count($bodyArray);$i=$i+count($headArray))
        {
            $table = $table. $rowStart;
            for($j = $i; $j < $i + count($headArray); $j++)
            {
                $table = $table. $cellData;
                $table = str_replace("{{value}}", $bodyArray[$j], $table);
            }
            $table = $table. $rowEnd;
        }
        $table = $table. $tableEnd;
        return $table;
    }

    public static function createSelect($name,$valuesArray)
    {
     $select = file_get_contents(__DIR__ . "/templates/select/selectHeader.html");
     $select = str_replace("{{selectName}}",$name,$select);
     for($i = 0; $i<count($valuesArray); $i++)
       {
         $select = $select . file_get_contents(__DIR__ . "/templates/select/selectBody.html");
         $select = str_replace("{{optionValue}}",$valuesArray[$i],$select);
       }
     $select = $select . file_get_contents(__DIR__ . "/templates/select/selectFooter.html");
     return $select;
    }
}
